Tier support for Donations wizard

claudiulodro opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

This PR adds support for tiers to the donations wizard:

![localhost_wp-admin_admin php_page=newspack-donations-wizard (1)](https://user-images.githubusercontent.com/7317227/61741155-1c42cf80-ad45-11e9-946f-79f37efb7781.png)
![localhost_wp-admin_admin php_page=newspack-donations-wizard (2)](https://user-images.githubusercontent.com/7317227/61741157-1e0c9300-ad45-11e9-9321-c7e4cd5f3bd4.png)

The donation block should be able to get all the info it needs through the `/newspack/v1/wizard/newspack-donations-wizard/donation` endpoint, including whether tiers are used and the tier settings.

### How to test the changes in this Pull Request:

1. Go to Donations wizard. Test that the tier-based and non-tier-based donation settings save/load well. You won't be able to see the low/high tiers on the product frontend, but the middle tier should be the  suggested price.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

jeffersonrabb commented on includes/configuration_managers/class-woocommerce-configuration-manager.php:
```diff
@@ -170,21 +218,23 @@ protected function create_donation_product( $args ) {
+		$defaults = $this->get_donation_default_settings();
 		$args = wp_parse_args( $args, $defaults );
 		$product_id = get_option( self::DONATION_PRODUCT_ID_OPTION, 0 );
 		$parent_product = \wc_get_product( $product_id );
```
I've been taking a slightly different approach to what goes in the configuration manager vs. the wizard class. Any statement that contains the plugin's classname goes in the configuration manager (same for setting plugin options) but everything else goes in the wizard class. I'm honestly not sure if that approach breaks down here, with such a mix of WC methods and utilities, but there might be some ways to separate the concerns. My favorite one so far is `Newspack Ads` — even though we control both sides of the equation the config manager methods are ultra simple: https://github.com/Automattic/newspack-plugin/blob/master/includes/configuration_managers/class-newspack-ads-configuration-manager.php. Part of the rationale for this is that these configuration manager classes are going to be the most brittle part of Newspack—prone to breakage whenever a plugin update is released. Even though we're not doing it yet, I'd like to build a robust suite of tests that download the latest version of each plugin and run though all config manager methods. This will be an effective way to quickly catch issues when something changes in the 3rd party plugin, and the simpler the config manager methods are the cleaner the tests will be.

claudiulodro replied:
In the interest of not getting too lost in the weeds, I'll leave it for now, and we can refactor in the future. I like the idea of having a test suite for our configuration managers to catch issues when plugins update; those are definitely going to be our most fragile classes.

sonjaleix commented:
@claudiulodro This is looking good, but I would change the order and hierarchy of this new feature. Currently it looks like the suggested donation amount is dependent on the tier toggle because the amount field is indented. 
<img width="235" alt="Screenshot 2019-07-24 15 08 23" src="https://user-images.githubusercontent.com/4823744/61828938-158b8980-ae25-11e9-82b7-ed47413df3ef.png">

**Recommended update:** 
<img width="446" alt="Screenshot 2019-07-24 15 10 17" src="https://user-images.githubusercontent.com/4823744/61829155-90ed3b00-ae25-11e9-9b9e-60334d9f341d.png">
<img width="446" alt="Screenshot 2019-07-24 15 10 17" src="https://user-images.githubusercontent.com/4823744/61829261-c2fe9d00-ae25-11e9-8bf5-42a3e5f4d8c9.png">

sonjaleix commented:
> It would be nice if we could add the currency symbol. I believe this came up in some of the feedback @sonjaleix received.

There is a Muriel design for the currency symbol in the text field. I added it to the new mockups of this feature in [Figma here](https://www.dropbox.com/s/d61fchrmb6kpf3t/Screenshot%202019-07-24%2015.32.53.png?dl=0).
<img width="453" alt="Screenshot 2019-07-24 15 32 53" src="https://user-images.githubusercontent.com/4823744/61830421-59cc5900-ae28-11e9-9339-f32009212424.png">

claudiulodro commented on assets/wizards/donations/views/donationSettingsScreen.js:
```diff
@@ -38,16 +38,56 @@ class DonationSettingsScreen extends Component {
+					checked={ tiered }
+					onChange={ tiered => onChange( 'tiered', tiered ) }
+				>
+					<div className='newspack-donations-wizard__tier-suggested-prices'>
```
Since they're just suggested prices, I don't think it makes sense to name the tiers.

sonjaleix replied:
> Since they're just suggested prices, I don't think it makes sense to name the tiers.
Agreed. For donations we should keep it simple.

claudiulodro commented on assets/wizards/donations/index.js:
```diff
@@ -31,8 +31,11 @@ class DonationsWizard extends Component {
 		super( ...arguments );
 		this.state = {
 			name: '',
+			suggestedAmountLow: 7.50,
```
Yeah, I think your way is better. Done as part of https://github.com/Automattic/newspack-plugin/pull/199/commits/71ac14791127d139c83cd5654c19960a49267997

claudiulodro commented on assets/wizards/donations/views/donationSettingsScreen.js:
```diff
@@ -38,16 +38,56 @@ class DonationSettingsScreen extends Component {
+							step="0.01"
+							label={ __( 'Suggested donation amount per month' ) }
+							value={ suggestedAmount }
+							onChange={ value => { 
```
`suggestedAmountUntiered` seems like a good solution; that way we won't have any surprising behavior. Done as part of https://github.com/Automattic/newspack-plugin/pull/199/commits/71ac14791127d139c83cd5654c19960a49267997

claudiulodro commented on assets/wizards/donations/views/donationSettingsScreen.js:
```diff
@@ -38,16 +38,56 @@ class DonationSettingsScreen extends Component {
+				</ToggleGroup>
+				{ ! tiered && (
+					<div className='newspack-donations-wizard__suggested-price'>
+						<TextControl
```
https://github.com/Automattic/newspack-plugin/pull/199/commits/e79f842fd1411b8cbac09f753232bda5a8362e02

claudiulodro commented on assets/wizards/donations/style.scss:
```diff
@@ -5,22 +5,22 @@
 
-	.muriel-info-button {
-		top: 16px;
+	> div {
```
I've updated this to follow Sonja's designs in https://github.com/Automattic/newspack-plugin/pull/199/commits/cc2458b8813f8cfc87084781762faae4178dd736 (This keeps the wide input on non-tiered)

claudiulodro commented:
I've made the following requested changes:
- Improved the data handling on the backend.
- Added currency preview to the inputs.
- Moved the toggle underneath the inputs.

claudiulodro commented:
Moved donations stuff into its own class in https://github.com/Automattic/newspack-plugin/pull/199/commits/42e394cb9675eb46c1349a0e2581695c43b68df5

Looks like I need to get signed into GH on my new computer, though :)

jeffersonrabb commented:
Note: rebased...

claudiulodro commented:
Thanks for reviewing!
